test/cpp/lazy/test_backend_device.cpp

```diff
@@ -20,8 +20,9 @@ TEST(BackendDeviceTest, Basic1) {
   auto device = BackendDevice();
 
   EXPECT_EQ(device.type(), 0);
-  EXPECT_EQ(device.ordinal(), 0);
-  EXPECT_STREQ(device.toString().c_str(), "Unknown0");
+  EXPECT_EQ(device.ordinal(), -1);
+  EXPECT_FALSE(device.is_valid());
+  EXPECT_STREQ(device.toString().c_str(), "Unknown-1");
```

Comment:
Just FYI, the `toString` response with default ordinal (-1) now becomes `Unknown-1` (this may be a bit confusing because the negative can be interpreted as a hyphen).

Reply:
Can you check what torch.device get printed in python by default? We can follow the convention there.

Reply:
Good point! So I've updated to follow the `c10::Device::str()` -- https://github.com/pytorch/pytorch/blob/master/c10/core/Device.cpp#L135. Ordinal only gets appended to the string if it exists.